### PR TITLE
Add new impulse 255, which gives you every weapon, just for debugging

### PR DESCRIPTION
**Pull request by a1batross**
Why 255? 0xFF is just a beautyful number.

### CODE REVIEW

File: regamedll/dlls/player.cpp [hunk 1]
```diff
@@ -6008,6 +6008,16 @@ void CBasePlayer::CheatImpulseCommands(int iImpulse)
...
 
+#ifdef REGAMEDLL_ADD
+	auto GiveFilledWeapon = [&](char* pszWeaponName) {
+		GiveNamedItem(pszWeaponName);
+		const WeaponInfoStruct *pInfo = GetWeaponInfo(pszWeaponName);
+		if (pInfo) {
+			GiveAmmo(pInfo->maxRounds, pInfo->ammoName2);
+		}
+	};
+#endif // REGAMEDLL_ADD
```
Comment (WPMGPRoSToTeMa) [inline regamedll/dlls/player.cpp]: Maybe create a block inside `case 255:` and move this lambda into it?
```C++
case 255: {
	auto GiveFilledWeapon = [&](char* pszWeaponName) {
		GiveNamedItem(pszWeaponName);
		const WeaponInfoStruct *pInfo = GetWeaponInfo(pszWeaponName);
		if (pInfo) {
			GiveAmmo(pInfo->maxRounds, pInfo->ammoName2);
		}
	};

	GiveFilledWeapon( "weapon_ak47" );
	GiveFilledWeapon( "weapon_aug" );
	// ...
}
```

Comment (a1batross) [reply regamedll/dlls/player.cpp]: Yeah, indeed. That should be better.

File: regamedll/dlls/player.cpp [hunk 2]
```diff
@@ -6027,6 +6027,47 @@ void CBasePlayer::CheatImpulseCommands(int iImpulse)
 			AddAccount(16000);
 			ALERT(at_console, "Crediting %s with $16000\n", STRING(pev->netname));
 			break;
+#ifdef REGAMEDLL_ADD
+		case 255: 
+		{
+			auto GiveFilledWeapon = [&](char* pszWeaponName) {
```
Comment (WPMGPRoSToTeMa) [inline regamedll/dlls/player.cpp]:
```suggestion
			const auto GiveFilledWeapon = [&](const char *pszWeaponName) {
```
and maybe change `GiveFilledWeapon` to `giveFilledWeapon`? (since it is a local variable)

File: regamedll/dlls/player.cpp [hunk 3]
```diff
@@ -6027,6 +6027,47 @@ void CBasePlayer::CheatImpulseCommands(int iImpulse)
...
+				GiveNamedItem(pszWeaponName);
+				const WeaponInfoStruct *pInfo = GetWeaponInfo(pszWeaponName);
+				if (pInfo) {
+					GiveAmmo(pInfo->maxRounds, pInfo->ammoName2);
+				}
+			};
+
+			GiveFilledWeapon( "weapon_ak47" );
```
Comment (WPMGPRoSToTeMa) [inline regamedll/dlls/player.cpp]: Remove spaces around parentheses?

File: regamedll/dlls/player.cpp [hunk 4]
```diff
@@ -4521,7 +4521,7 @@ void CBasePlayer::UpdateGeigerCounter()
 	m_flgeigerDelay = gpGlobals->time + 0.25;
 
 	// send range to radition source to client
-	range = byte(m_flgeigerRange / 4.0);//* 0.25);		// TODO: ACHECK!
+	range = byte(m_flgeigerRange / 4.0);// * 0.25);		// TODO: ACHECK!
```
Comment (WPMGPRoSToTeMa) [inline regamedll/dlls/player.cpp]: Unrelated change.

File: regamedll/dlls/player.cpp [hunk 5]
```diff
@@ -6027,6 +6027,48 @@ void CBasePlayer::CheatImpulseCommands(int iImpulse)
 			AddAccount(16000);
 			ALERT(at_console, "Crediting %s with $16000\n", STRING(pev->netname));
 			break;
+#ifdef REGAMEDLL_ADD
+		case 255:
+		{
+			auto giveFilledWeapon = [&](const char* pszWeaponName) {
```
Comment (WPMGPRoSToTeMa) [inline regamedll/dlls/player.cpp]: `const auto` doesn't fit here?

Comment (WPMGPRoSToTeMa) [inline regamedll/dlls/player.cpp]: `const char *pszWeaponName` please.

Comment (WPMGPRoSToTeMa) [inline regamedll/dlls/player.cpp]: Maybe rename `pszWeaponName` to `weaponName`?

File: regamedll/dlls/player.cpp [hunk 6]
```diff
@@ -6027,6 +6027,48 @@ void CBasePlayer::CheatImpulseCommands(int iImpulse)
...
+				GiveNamedItem(pszWeaponName);
+				const WeaponInfoStruct *pInfo = GetWeaponInfo(pszWeaponName);
```
Comment (WPMGPRoSToTeMa) [inline regamedll/dlls/player.cpp]: Consider `const WeaponInfoStruct &info`, so `info.maxRounds` and `info.ammoName2`.

File: regamedll/dlls/player.cpp (hunk 4 above)
Comment (a1batross) [reply regamedll/dlls/player.cpp]: It fixes code highlighting in GNU Nano. 

I did not mentioned it, as it just a comment.

File: regamedll/dlls/player.cpp (hunk 5 above)
Comment (a1batross) [reply regamedll/dlls/player.cpp]: Don't think so. `auto` main feature is to make code shorter, in this case `const` it doesn't make sense, as it's just a lambda.

File: regamedll/dlls/player.cpp (hunk 6 above)
Comment (a1batross) [reply regamedll/dlls/player.cpp]: GetWeaponInfo return value intended to be a pointer, isn't it?

File: regamedll/dlls/player.cpp (hunk 5 above)
Comment (WPMGPRoSToTeMa) [reply regamedll/dlls/player.cpp]: @a1batross and even without `const` this variable can't be reassigned since lambdas with capture don't have assignment operator (but assignment operator is available for stateless lambdas in C++20).

File: regamedll/dlls/player.cpp (hunk 4 above)
Comment (WPMGPRoSToTeMa) [reply regamedll/dlls/player.cpp]: @a1batross can you move this change to a new PR?

File: regamedll/dlls/player.cpp (hunk 6 above)
Comment (WPMGPRoSToTeMa) [reply regamedll/dlls/player.cpp]: Yes, but you can "convert" it to a reference by dereferencing it.

Comment (a1batross) [reply regamedll/dlls/player.cpp]: And still, check for null pointer? %)

Comment (WPMGPRoSToTeMa) [reply regamedll/dlls/player.cpp]: @a1batross can it return `nullptr` for this code?
